File: back/microsservicos/aluguel/router.ts
import { Router } from "express";

import { createAluguelController } from "./app/create_aluguel/create_aluguel_presenter";
import { getAllAluguelController } from "./app/get_all_aluguel/get_all_aluguel_presenter";
import { getAluguelController } from "./app/get_aluguel/get_aluguel_presenter";
import { updateAluguelController } from "./app/update_aluguel/update_aluguel_presenter";
import { deleteAluguelController } from "./app/delete_aluguel/delete_aluguel_presenter";

const router: Router = Router()

router.post('/aluguel', (req, res) => createAluguelController.handle(req, res))
router.get('/aluguel', (req, res) => getAllAluguelController.handle(req, res))
router.get('/aluguel/:id', (req, res) => getAluguelController.handle(req, res))
router.patch('/aluguel/:id', (req, res) => updateAluguelController.handle(req, res))
router.delete('/aluguel/:id', (req, res) => deleteAluguelController.handle(req, res))

export { router }

File: back/microsservicos/aluguel/lib/presentation/http/router.dart
import 'package:aluguel_dart/presentation/http/controllers/create_aluguel_controller.dart';
import 'package:aluguel_dart/presentation/http/controllers/delete_aluguel_controller.dart';
import 'package:aluguel_dart/presentation/http/controllers/get_all_aluguel_controller.dart';
import 'package:aluguel_dart/presentation/http/controllers/get_aluguel_controller.dart';
import 'package:aluguel_dart/presentation/http/controllers/get_door_hash_controller.dart';
import 'package:aluguel_dart/presentation/http/controllers/update_aluguel_controller.dart';
import 'package:shelf/shelf.dart';
import 'package:shelf_router/shelf_router.dart';


Router buildAluguelRoutes({
  required GetAluguelController getController,
  required CreateAluguelController createController,
  required UpdateAluguelController updateController,
  required DeleteAluguelController deleteController,
  required GetAllAluguelController getAllAluguelController,
  required GetDoorHashController getDoorHashController,
}) {
  final r = Router();

  // Health
  r.get('/health', (_) => Response.ok('ok'));

  r.get('/aluguel', getController.handle);

  r.get('/all-aluguel', getAllAluguelController.handle);

  r.post('/aluguel', createController.handle);

  r.patch('/aluguel', updateController.handle);

  r.delete('/aluguel', deleteController.handle);

  r.post('/get-door-hash', getDoorHashController.handle);

  return r;
}
